Cannot install on Windows

The README suggests that installation is as simple as `pip install mar`, but this does not seem to work on my Windows 11 machine. The problem seems to lie with the dependency `backports.lzma`. I cannot install it independently either. It seems that it wants to compile some code that depends on `lzma.h`, but my system has no such library and it isn't clear how I can obtain one. This is what happens specifically in the installation process:

```
> pip install mar
Collecting mar
  Using cached mar-3.1.0-py2.py3-none-any.whl (30 kB)
Requirement already satisfied: click in c:\mozilla-build\python3\lib\site-packages (from mar) (8.1.3)
Collecting construct
  Using cached construct-2.10.68.tar.gz (57 kB)
  Preparing metadata (setup.py) ... done
Collecting asn1crypto
  Using cached asn1crypto-1.5.1-py2.py3-none-any.whl (105 kB)
Collecting backports.lzma
  Using cached backports.lzma-0.0.14.tar.gz (47 kB)
  Preparing metadata (setup.py) ... done
Collecting cryptography
  Using cached cryptography-37.0.4-cp36-abi3-win_amd64.whl (2.4 MB)
Requirement already satisfied: colorama in c:\mozilla-build\python3\lib\site-packages (from click->mar) (0.4.5)
Requirement already satisfied: cffi>=1.12 in c:\mozilla-build\python3\lib\site-packages (from cryptography->mar) (1.15.1)
Requirement already satisfied: pycparser in c:\mozilla-build\python3\lib\site-packages (from cffi>=1.12->cryptography->mar) (2.21)
Using legacy 'setup.py install' for backports.lzma, since package 'wheel' is not installed.
Using legacy 'setup.py install' for construct, since package 'wheel' is not installed.
Installing collected packages: backports.lzma, asn1crypto, construct, cryptography, mar
  Running setup.py install for backports.lzma ... error
  error: subprocess-exited-with-error

  × Running setup.py install for backports.lzma did not run successfully.
  │ exit code: 1
  ╰─> [24 lines of output]
      This is backports.lzma version 0.0.14
      running install
      C:\mozilla-build\python3\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-39
      creating build\lib.win-amd64-cpython-39\backports
      copying backports\__init__.py -> build\lib.win-amd64-cpython-39\backports
      creating build\lib.win-amd64-cpython-39\backports\lzma
      copying backports\lzma\__init__.py -> build\lib.win-amd64-cpython-39\backports\lzma
      running build_ext
      building 'backports.lzma._lzma' extension
      creating build\temp.win-amd64-cpython-39
      creating build\temp.win-amd64-cpython-39\Release
      creating build\temp.win-amd64-cpython-39\Release\backports
      creating build\temp.win-amd64-cpython-39\Release\backports\lzma
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\mozilla-build\python3\include -IC:\Users\bytesized\include -I/opt/local/include -I/usr/local/include -IC:\mozilla-build\python3\include -IC:\mozilla-build\python3\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\cppwinrt" /Tcbackports/lzma/_lzmamodule.c /Fobuild\temp.win-amd64-cpython-39\Release\backports/lzma/_lzmamodule.obj
      _lzmamodule.c
      backports/lzma/_lzmamodule.c(24): warning C4273: 'PyErr_NewExceptionWithDoc': inconsistent dll linkage
      C:\mozilla-build\python3\include\pyerrors.h(225): note: see previous definition of 'PyErr_NewExceptionWithDoc'
      backports/lzma/_lzmamodule.c(115): fatal error C1083: Cannot open include file: 'lzma.h': No such file or directory
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.32.31326\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> backports.lzma

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

Eventually I just opened a Linux VM where the installation seemed to work just fine.

I initially considered opening an issue with the [`backports.lzma` repo](https://github.com/peterjc/backports.lzma/issues), but it seems to be archived now so I cannot.

--- a/setup.py
+++ b/setup.py
@@ -9,6 +9,7 @@ from __future__ import print_function
 
 import io
 import re
+import sys
 from glob import glob
 from os.path import basename
 from os.path import dirname
@@ -17,6 +18,16 @@ from os.path import splitext
 
 from setuptools import find_packages
 from setuptools import setup
+
+
+requires = [
+    'asn1crypto',
+    'click',
+    'construct',
+    'cryptography',
+]
+if sys.version_info[0] == 2:
+    requires.append('backports.lzma')
 
 
 def read(*names, **kwargs):
@@ -69,13 +80,7 @@ setup(
     keywords=[
         'mozilla', 'mar', 'archive',
     ],
-    install_requires=[
-        'asn1crypto',
-        'backports.lzma',
-        'click',
-        'construct',
-        'cryptography',
-    ],
+    install_requires=requires,
     extras_require={
         # eg:
         #   'rst': ['docutils>=0.11'],

